### ISSUE TITLE
API request: xds async NACK

### ISSUE THREAD

**Issue by mandarjog** [envoyproxy/envoy#9447]
When envoy receives a listener from management server it has 3 choices
1. accept listener and put it in ready state
2. Reject listener with a NACK
3. Accept config and put listener in a warming state awaiting an async operation.

If the async operation fails for any reason and it was performed against separate management server, the original management server has no way to know what happened.

There should be a way to indicate a NACK after async operations complete.

This happens with SDS, Wasm code fetch via url or fcds.

Related issues:
https://github.com/envoyproxy/envoy/issues/5739
https://github.com/envoyproxy/envoy/issues/7867

**Comment by htuch** [comment]
I agree with the problem statement. In terms of solution space, we've also discussed a config status reverse subscription, which I think wold be more aligned with where we are heading in UDPA-TP than a dedicated NACK message. @costinm has also expressed interest here.

**Comment by mattklein123** [comment]
> I agree with the problem statement. In terms of solution space, we've also discussed a config status reverse subscription, which I think wold be more aligned with where we are heading in UDPA-TP than a dedicated NACK message. 

+1. My fear, just speaking about the Envoy implementation, is that if we try to send a delayed NACK if something fails async it's going to be incredibly complicated.

**Comment by costinm** [comment]
IMHO xDS acts very close to a pubsub system: you subscribe to a topic (typeURL) and get some messages. 

Instead of introducing a 'delayed NACK' concept - which I agree would be quite complicated - we can just define a new resource type - '.../config_status' or '/nacks' - and have envoy ( or any XDS client ) publish on the topic, assuming that #11396 is accepted. 

The nice thing about treating it as 'pubsub' is that the published NACK (or any message originated from envoy or its extensions/WASM) can be bridged into pubsub systems and handled by automated systems - instead of just a complicated protocol detail between envoy and XDS server.
